### DISCUSSION TITLE
Support building all types of Statement into string through a trait

### DISCUSSION THREAD

**Discussion by siketyan** [SeaQL/sea-query#526] [Ideas]
Related thread on Discord: https://discord.com/channels/873880840487206962/1045022584162500728

## The problem

SeaORM can query their managed entities using a query generated by SeaQuery, but that needs the following code to be written every time:

```rust
let backend = db.get_database_backend();
let statement = Statement::from_string(backend, query.to_string(backend.get_query_builder()));
```

This is a little complicated, so I am consindering to utilise this, for example:

```rust
let backend = db.get_database_backend();
let statement = Statement::from_query(backend, query);
```

## Implementation

To achieve this, we should modify both SeaORM and SeaQuery.

First, *Statement structs such as [SelectStatement](https://docs.rs/sea-query/latest/sea_query/query/struct.SelectStatement.html) has a method `to_string`, but this is not in any trait.
It's ok to just implement the SeaQuery to SeaORM conversion, but SeaORM's dependency to SeaQuery here will be much compilcated.
To avoid this and to help future extensions, I think we should define a new trait in SeaQuery and implement it on *Statement structs:

```rust
pub trait IntoQuery {
    pub fn into_query<T: QueryBuilder>(&self, query_builder: T) -> String;
}
```

Next, we need to add a method on `Statement` like `Statement::from_query`.
I am considering the following signature for it:

```rust
pub fn from_query<Q: IntoQuery>(db_backend: DbBackend, query: Q) -> Self;
```

Any idea on this?
I am going to work on this if this idea were agreed from the community and/or contributors.

Thank you for reading!

**Comment by billy1624** [comment]
Hey @siketyan, thanks for asking!! I think there is a better approach than that?
Check https://www.sea-ql.org/sea-orm-cookbook/003-run-sea-query-statement-on-sea-orm.html

**Reply by siketyan** [reply]
@billy1624 
Oh, I did not notice that 😔 Thank you for suggesting that.
As Stormblessed16 said on Discord, it would be great if that is documented in https://www.sea-ql.org/SeaORM/docs/basic-crud/raw-sql/ or somewhere I think.

**Reply by billy1624** [reply]
Would you be interested in editing the documentation? An PR would be appreciated!!

**Reply by siketyan** [reply]
Of course it's okay, I will open a PR later

**Reply by billy1624** [reply]
Thanks!! @siketyan
Let me know if you need help
